pkg/v1/mutate/mutate.go

```diff
@@ -392,3 +393,73 @@ func inWhiteoutDir(fileMap map[string]bool, file string) bool {
 	}
 	return false
 }
+
+// StripConfig strips out any timestamp from the layers and config of an image.
+func StripConfig(img v1.Image) (v1.Image, error) {
+	manifest, err := img.Manifest()
+	if err != nil {
+		return nil, err
+	}
+
+	layers, err := img.Layers()
+	if err != nil {
+		return nil, err
+	}
+
+	var newLayers []v1.Descriptor
+	var newDiffIDs []string
+	for _, layer := range layers {
+		newLayer, err := stripLayer(layer)
+		if err != nil {
+			return nil, err
+		}
+		newLayerName, err := newLayer.Digest()
+		if err != nil {
+			return nil, err
+		}
+
+		layerDescriptor := &v1.Descriptor{
+			MediaType: types.DockerLayer,
+		}
+		newLayers = append(newLayers, *layerDescriptor)
```

Comment:
I think you may need to make a new image from scratch: https://github.com/google/go-containerregistry/blob/master/pkg/v1/empty/image.go

And then use mutate to append the new stripped layers into it.

Then use mutate.Config to copy the Config into the new image.